### PR TITLE
Streaming AudioManager

### PR DESCRIPTION
**Pull request by longtomjr**
Created a Manager for audio in pp.   
Audio files are now loaded from StreamingAssets/Audio  

Thanks for everyone in discord helping out!
Special Thanks to @powli for helping out by changing my code to mostly working condition.

### CODE REVIEW

File: Assets/Scripts/Controllers/AudioManager.cs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+// ====================================================
+#endregion
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using UnityEngine;
+
+/// <summary>
+/// The Manager that handles the loading and storing of audio form streamingAssets.
```
Comment (mikejbrown) [inline Assets/Scripts/Controllers/AudioManager.cs]: from

File: Assets/Scripts/Controllers/AudioManager.cs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+            {
+                Debug.LogWarning("No audio available called: " + audioNameAndCategory);
+                clip = audioClips["Sound/Error.ogg"];
+            }
+            catch
+            {
+                throw new FileNotFoundException("Sound/Error.ogg not found");
+            }
+
+        }
```
Comment (mikejbrown) [inline Assets/Scripts/Controllers/AudioManager.cs]: Stylecop `SA1508 : CSharp.Layout : A closing curly bracket must not be preceded by a blank line. (Assembly-CSharp)`

File: Assets/Scripts/Controllers/AudioManager.cs [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        {
+            LoadAudioFiles(subDirectory);
+        }
+
+        string[] filesInDir = Directory.GetFiles(directoryPath);
+        IEnumerable<WWW> audioFiles = LoadAudioFile(filesInDir, directoryPath);
+        IEnumerator<WWW> e = audioFiles.GetEnumerator();
+        while (e.MoveNext())
+        {
+            Debug.Log("Type: " + e.Current.GetType());
```
Comment (mikejbrown) [inline Assets/Scripts/Controllers/AudioManager.cs]: log spam

File: Assets/Scripts/Controllers/AudioManager.cs [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+            string filePath = new FileInfo(file).FullName;
+
+            if (filePath.Contains(".xml") || filePath.Contains(".meta") || filePath.Contains(".db"))
+            {
+                continue;
+            }
+
+            WWW www = new WWW(@"file://" + filePath);
+
+            Debug.Log("before - " + www.error);
```
Comment (mikejbrown) [inline Assets/Scripts/Controllers/AudioManager.cs]: log spam (only report if there is an actual error)

File: Assets/Scripts/Controllers/AudioManager.cs [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+            yield return www;
+
+            Debug.Log("After - " + www.error);
```
Comment (mikejbrown) [inline Assets/Scripts/Controllers/AudioManager.cs]: log spam

File: Assets/Scripts/Controllers/AudioManager.cs [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+
+            AudioClip clip = www.GetAudioClip(false, false);
+
+            Debug.Log(clip.name + "Downloaded");
```
Comment (mikejbrown) [inline Assets/Scripts/Controllers/AudioManager.cs]: `clip.name` doesn't seem to be populated in my testing?? But one informative log message per file is all you need. Would prefer something like
`Debug.ULogChannel("AudioManager", "Audio file loaded: '{0}'.", clip.name)`, assuming `clip.name` was set properly.

Comment (mikejbrown) [inline Assets/Scripts/Controllers/AudioManager.cs]: Make that `filePath` for the log message instead of `clip.name`.

**Comment by koosemose** [comment]
Conflicts

**Comment by longtomjr** [comment]
Resolved all conflicts

**Pull request merged**